### PR TITLE
feat: fixing main deployment issues

### LINKED ISSUE #630: Parking Sensor (Databricks) deployment fails due to change in behaviour of workspace startup
## Description

Deployment fails with `parse error: Invalid numeric literal at line 1, column 6".`

```
++ DATABRICKS_HOST=https://adb-<workspace_id>.xx.azuredatabricks.net
++ bash -c 'databricks tokens create --comment '\''deployment'\'''
++ jq -r .token_value
parse error: Invalid numeric literal at line 1, column 6
+ databricks_token=
```
**Root Cause:** 
The error message is a generic error that hides the real error message of "Error: Authorization failed. Your token may be expired or lack the valid scope". 

This appears to be a cold start problem with the Azure Databricks workspace. When deploying an Azure Databricks workspace, the workspace needs to be manually launched first in the Azure portal before the API calls are permitted. 

**Workaround:**

When you encounter the error, manually launch the Databricks workspace, then re-run the deployment script (`deploy.sh`). Ensure you have set the `DEPLOYMENT_ID` environment variable. The script is designed to be idempotent to each deployment determined by the `DEPLOYMENT_ID`.

Note, if you don't set the `DEPLOYMENT_ID` variable, the `deploy.sh` script will simply create a fresh deployment with an random deployment_id which will fail instead of targeting your previous deployment.

## Expected Behavior

Parking Sensor (databricks) deploys successfully.

## Reproduce

Deploy the Parking Sensor Databricks as per instructions.

## Additional Context

- If applicable: Add any other context about the problem here; for example: doc changes, screenshots, logs, links, etc

### LINKED ISSUE #763: Upgrade Databricks CLI
The current sample uses the Databricks CLI (Legacy) version that won't be supported moving forward. Check here for more details: https://docs.databricks.com/ja/archive/dev-tools/cli/index.html. The E2E sample needs to be upgraded to install and use the new CLI - check for more details here: https://docs.databricks.com/en/dev-tools/cli/install.html

A Warning message is thrown during the deployment: 
WARN: Your CLI is configured to use Jobs API 2.0. In order to use the latest Jobs features please upgrade to 2.1: 'databricks jobs configure --version=2.1'. Future versions of this CLI will default to the new Jobs API. Learn more at https://docs.databricks.com/dev-tools/cli/jobs-cli.html

**DoD**
- Modify deployment to install latest Databricks CLI
- Modify bash scripts using the legacy CLI to the new commands available in the new CLI
- Adjust documentation accordingly

### LINKED ISSUE #778: Fixing Dedicated Synapse pool deployment
Error while deploying the E2E sample
'Failed'.","details":[{"code":"DeploymentFailed","target":"/subscriptions/SUBCRIPTION/resourceGroups/RESOURCE_GROUP/providers/Microsoft.Resources/deployments/SYNAPSE_POOL","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.","details":[{"code":"InvalidServerAADOnlyAuthNoAADAdminPropertyName","message":"AAD Admin is not configured, AAD Admin must be set before enabling/disabling AAD Only Authentication."}]}]}]}}

**DoD**
- Deploy Synapse Pool without errors

### PR DESCRIPTION
**Pull request by ydaponte**
# Type of PR

- Documentation changes
- Code changes

## Purpose
It fixes the following issues:
- Failure due to change in behavior of workspace startup, instead there is a pause in the script to authenticate to the databricks workspace
- Dedicated Synapse pool deployment failure
- Upgrade Databricks CLI as the Databricks configuration script was failing with the previous CLI

## Does this introduce a breaking change? If yes, details on what can break
It might introduce a breaking change as the Databricks CLI is being upgraded. 
The deployment needs to be tested end-to-end.

## Author pre-publish checklist
- [ ] Added test to prove my fix is effective or new feature works
- [X] No PII in logs
- [X] Made corresponding changes to the documentation

## Validation steps
- Clone the repository
- Launch the devcontainer
- Verify within the devcontainer that the databricks CLI version is Databricks CLI v0.232.1 or above by running databricks --version
- Follow the pre-requirements to deploy the sample
- Run the ./deploy.sh script
- Verify that the deployment is successful in the 3 environments
- Within the databricks workspaces, verify that the cluster was created successfully (including the ddo library), and that the job executes successfully - all the tables are created under the Hive Metastore

## Issues Closed or Referenced

- Closes #630 
- Closes #778 
- Closes #763

### CODE REVIEW

File: e2e_samples/parking_sensors/.devcontainer/post_create.sh [hunk 1]
```diff
@@ -4,4 +4,8 @@
 az devops configure --defaults organization="$AZDO_ORGANIZATION_URL" project="$AZDO_PROJECT"
 
 # Install requirements depending if devcontainer was openned at root or in parking_sensor folder.
-pip install -r /workspace/e2e_samples/parking_sensors/src/ddo_transform/requirements_dev.txt
+pip install -r /workspace/e2e_samples/parking_sensors/src/ddo_transform/requirements_dev.tx
```
Comment (thesqlpro) [inline e2e_samples/parking_sensors/.devcontainer/post_create.sh:7]: you changed the file extension from .txt to tx is this correct?

**Review by thesqlpro — APPROVED** [review]
Just one comment on a file extension change
Looks good otherwise

**Review by bsherwin — APPROVED** [review]
Looks Good To Me... LGTM!

File: e2e_samples/parking_sensors/.devcontainer/Dockerfile [hunk 2]
```diff
@@ -45,6 +45,9 @@ RUN apt-get update \
     temurin-21-jdk
 ENV JAVA_HOME=/usr/lib/jvm/temurin-21-jdk-amd64
 
+RUN [ -f /usr/local/bin/databricks ] && rm /usr/local/bin/databricks || echo "File not found, skipping removal"
+RUN curl -fsSL https://raw.githubusercontent.com/databricks/setup-cli/main/install.sh | sh
```
[the remark this answers is not in the document]
Comment (ydaponte) [reply e2e_samples/parking_sensors/.devcontainer/Dockerfile:49]: Yes, as agreed! Good point. Let's plan to change it as a priority.

**Pull request merged**